Library/Homebrew/cmd/update-bash.sh

```diff
@@ -153,38 +199,37 @@ pull() {
   fi
 
   INITIAL_REVISION="$(read_current_revision)"
+  export HOMEBREW_UPDATE_BEFORE"$TAP_VAR"="$INITIAL_REVISION"
 
   # ensure we don't munge line endings on checkout
   git config core.autocrlf false
 
-  trap reset_on_interrupt SIGINT
-
-  if [ -n "$HOMEBREW_REBASE" ]
+  if [[ -n "$HOMEBREW_REBASE" ]]
   then
     git rebase $QUIET_ARGS "origin/$UPSTREAM_BRANCH"
   else
     git merge --no-edit --ff $QUIET_ARGS "origin/$UPSTREAM_BRANCH"
   fi
 
-  trap - SIGINT
+  export HOMEBREW_UPDATE_AFTER"$TAP_VAR"="$(read_current_revision)"
 
-  CURRENT_REVISION="$(read_current_revision)"
-  export HOMEBREW_UPDATE_AFTER"$TAP_VAR"="$(git rev-parse "$UPSTREAM_BRANCH")"
+  trap '' SIGINT
 
-  if [ "$INITIAL_BRANCH" != "$UPSTREAM_BRANCH" ] && [ -n "$INITIAL_BRANCH" ]
+  if [[ "$INITIAL_BRANCH" != "$UPSTREAM_BRANCH" && -n "$INITIAL_BRANCH" ]]
   then
     git checkout "$INITIAL_BRANCH" $QUIET_ARGS
     pop_stash
   else
     pop_stash_message
   fi
+
+  trap - SIGINT
 }
 
 update-bash() {
-  if [ -z "$HOMEBREW_DEVELOPER" ]
+  if [[ -z "$HOMEBREW_DEVELOPER" ]]
   then
-    echo "This command is currently only for Homebrew developers' use." >&2
-    exit 1
+    odie "This command is currently only for Homebrew developers' use."
   fi
 
   for i in "$@"
```

Comment:
Can we make `i` a local variable? Similarly for any other variable that is only used locally but not declared as such.